src/ur/robot_receive_timeout.cpp

```diff
@@ -0,0 +1,116 @@
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+// -- END LICENSE BLOCK ------------------------------------------------
+
+#include "ur_client_library/ur/robot_receive_timeout.h"
+#include "ur_client_library/log.h"
+#include "ur_client_library/exceptions.h"
+#include "ur_client_library/types.h"
+
+#include <sstream>
+
+namespace urcl
+{
+RobotReceiveTimeout::RobotReceiveTimeout(std::chrono::milliseconds timeout) : timeout_(timeout)
+{
+}
+
+RobotReceiveTimeout RobotReceiveTimeout::millisec(const unsigned int milliseconds)
+{
+  std::chrono::milliseconds timeout = std::chrono::milliseconds(milliseconds);
+  RobotReceiveTimeout robot_receive_timeout(timeout);
+  return robot_receive_timeout;
+}
+
+RobotReceiveTimeout RobotReceiveTimeout::sec(const float seconds)
+{
+  std::chrono::milliseconds timeout = std::chrono::milliseconds(static_cast<int>(seconds * 1000));
+  RobotReceiveTimeout robot_receive_timeout(timeout);
+  return robot_receive_timeout;
+}
+
+RobotReceiveTimeout RobotReceiveTimeout::off()
+{
+  RobotReceiveTimeout robot_receive_timeout(std::chrono::milliseconds(0));
+  return robot_receive_timeout;
+}
+
+int RobotReceiveTimeout::verifyRobotReceiveTimeout(const comm::ControlMode control_mode,
+                                                   const std::chrono::milliseconds step_time) const
+{
+  // Convert timeout to float
+  if (comm::ControlModeTypes::is_control_mode_non_realtime(control_mode))
+  {
+    if (timeout_ < step_time && timeout_ > std::chrono::milliseconds(0))
+    {
+      std::stringstream ss;
+      ss << "Robot receive timeout " << timeout_.count() << " is below the step time " << step_time.count()
+         << ". It will be reset to the step time.";
+      URCL_LOG_ERROR(ss.str().c_str());
+      return step_time.count();
+    }
+    else
+    {
+      return timeout_.count();
+    }
+  }
+  else if (comm::ControlModeTypes::is_control_mode_realtime(control_mode))
+  {
+    const std::chrono::milliseconds max_realtime_timeout = std::chrono::milliseconds(1000);
+    if (timeout_ < step_time)
+    {
+      std::stringstream ss;
+      ss << "Realtime read timeout " << timeout_.count() << " is below the step time " << step_time.count()
+         << ". It will be reset to the step time.";
+      URCL_LOG_ERROR(ss.str().c_str());
+      return step_time.count();
+    }
+    else if (timeout_ > max_realtime_timeout)
+    {
+      std::stringstream ss;
+      ss << "Robot receive timeout " << timeout_.count()
+         << " is above the maximum allowed timeout for realtime commands " << max_realtime_timeout.count()
+         << ". It will be reset to the maximum allowed timeout.";
```

Comment:
```suggestion
         << "ms is above the maximum allowed timeout for realtime commands " << max_realtime_timeout.count()
         << "ms. It will be reset to the maximum allowed timeout.";
```